boxes/boxes/vanilla/app/embedded-wallet.ts

```diff
@@ -83,15 +84,21 @@ export class EmbeddedWallet extends BaseWallet {
       (await this.aztecNode.getCurrentBaseFees()).mul(1 + this.baseFeePadding);
     const sponsoredFPCContract =
       await EmbeddedWallet.#getSponsoredPFCContract();
-    const paymentMethod =
-      userFeeOptions?.paymentMethod ??
-      new SponsoredFeePaymentMethod(sponsoredFPCContract.instance.address);
+    const paymentMethod = !userFeeOptions?.embeddedPaymentMethodFeePayer
+      ? new SponsoredFeePaymentMethod(sponsoredFPCContract.instance.address)
+      : undefined;
```

Comment:
```suggestion
    const paymentMethod = userFeeOptions?.embeddedPaymentMethodFeePayer
      ? undefined : new SponsoredFeePaymentMethod(sponsoredFPCContract.instance.address);
```
Think it's more readable when not having to do the bool flip.

Could you add here a comment why we populate the payment method only when embeddedPaymentMethodFeePayer is undefined and why the sponsored fee payment method is used?

Reply:
This has evolved, but still I prefer the actual "something" return value first, I find it easier to read. Different brain damage.